src/firefly/js/ui/RangeSliderView.jsx

```diff
@@ -1,79 +1,51 @@
-import React, {PureComponent} from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import {get, has, isNumber, isString, isObject, isNaN} from 'lodash';
+import {has, isNumber, isString, isObject, isNaN} from 'lodash';
 import Slider from 'rc-slider';
 import './rc-slider.css';
 
-/**
- * @summary slider component
- */
-export class RangeSliderView extends PureComponent {
-    constructor(props) {
-        super(props);
-
-        this.state = {value: parseFloat(props.slideValue)};
-        this.onSliderChange = this.onSliderChange.bind(this);
-        const d = get(props, 'decimalDig', 3);
-        this.decimalDig = (d < 0) ? 0 : (d > 20) ? 20 : d;
-    }
-
-    static getDerivedStateFromProps(props,state) {
-        return {value: parseFloat(props.slideValue)};
-    }
-
-    onSliderChange(v) {
-        const {handleChange, minStop, maxStop} = this.props;
-
-        if (minStop && v < minStop ) {
-            v = minStop;
-        } else if (maxStop && v > maxStop) {
-            v = maxStop;
-        }
-
-        this.setState({value: v});
-        v = parseFloat(v.toFixed(this.decimalDig));
-
-        if (handleChange) {
-            handleChange(`${v}`);      //value could be any number of decimal
-        }
-    }
-
-
-    render() {
-        const {wrapperStyle={}, sliderStyle={}, className, marks, vertical,
-             defaultValue, handle, label, labelWidth, tooltip, minStop, maxStop, min, max, step} = this.props;
-        const {value} = this.state;    //displayValue in string, min, max, step: number
-        let val = parseFloat(value);
 
+export function RangeSliderView({min=0, max=100, minStop, maxStop, className, marks, step=1, vertical=false,
+                                    defaultValue=0, slideValue, handle, style={}, sliderStyle={},
+                                    label='', labelWidth, tooltip, decimalDig=3, handleChange}){
+    function toRangeBound(val) {
```

Comment:
Usually for closure functions we use the arrow notation (`const xx => ()`) and for top level (in a file) we use `function` notation.